### PR TITLE
refactor(alert, dropdown): `onToggleOpenCloseComponent` replaces connect logic

### PR DESCRIPTION
**Pull request by Elijbet**
**Related Issue:** #6018

## Summary
Use the new `onToggleOpenCloseComponent` pattern to replace both `connectOpenCloseComponent` and `disconnectOpenCloseComponent` on `alert` and `dropdown`. This new pattern is intended to simplify the overall eventing logic.

### CODE REVIEW

File: packages/calcite-components/src/components/alert/alert.tsx [hunk 1]
```diff
@@ -89,6 +85,7 @@ export class Alert implements OpenCloseComponent, LoadableComponent, T9nComponen
   openHandler(): void {
     if (this.open && !this.queued) {
       this.calciteInternalAlertRegister.emit();
+      onToggleOpenCloseComponent(this);
```
Comment (jcfranco) [inline packages/calcite-components/src/components/alert/alert.tsx:88]: This needs to be called when the component is both open and closed, right?

**Review by jcfranco — CHANGES_REQUESTED** [review]
Can you look at adding tests for these changes? This behavior hasn't been tested so far and I'd like to avoid any possible regressions from the refactor.

We could use them as the base for the test helper. 🎉

**Comment by Elijbet** [comment]
> Can you look at adding tests for these changes? This behavior hasn't been tested so far and I'd like to avoid any possible regressions from the refactor.
> 
> We could use them as the base for the test helper. 🎉

The behavior is whether the events are emitted right? 
eg.
  ```it("should emit component status for transition-chained events: 'calciteInputDatePickerBeforeOpen', 'calciteInputDatePickerOpen', 'calciteInputDatePickerBeforeClose', 'calciteInputDatePickerClose'", async () => {```

**Comment by jcfranco** [comment]
> The behavior to test is whether the events are emitted right?

Correct.

**Review by jcfranco — APPROVED** [review]
Awesome! Thanks for pointing out that the existing tests already covered this changes. 

Per our conversation, let's extract those tests into a test helper and update it to cover disabled animations.

**Pull request merged**